### PR TITLE
Add code of conduct for the team

### PR DESCRIPTION
**Pull request by malvikasharan**
# Summary

- Adding Code of Conduct for the team.
- Ask all members to review and leave a comment if they are happy to have this Code of Conduct added to the project
- Need the restorative justice element to be discussed and reviewed with @KirstieJane

### CODE REVIEW

**Comment by KirstieJane** [comment]
I really love the focus on restorative justice!

My only very high level feedback at the moment is that I think we need to be clear about how the code of conduct interacts - or where it sits in a pipeline - with Turing HR processes. (Or - if we make it more general, HR processes at any organisation.)

Probably my suggestion is that the code of conduct should be norm setting within a team _before_ any involvement of HR. Then - if an HR process, including an informal one, is started, that one needs to take precident over code of conduct enforcement. It's a bit sad because although HR will always _try_ for a restorative justice position they may not take the same steps "we" would if we were handling a complaint through the code of conduct alone. So being really clear where the boundaries are is the best - if imperfect - way forwards.

**Comment by malvikasharan** [comment]
@KirstieJane I have included your comment with appropriate Turing HR policy links in the claus 15.2. I will get the CM team review this and apply to this repo to avoid a near CoC breach incident that could occur on this repo (commenting blocked on issue by me).

**Comment by RaoOfPhysics** [comment]
Morning, @malvikasharan. I gave this a read yesterday and am happy from my side for it to be merged.

**Comment by arronlacey** [comment]
agree with @RaoOfPhysics - happy for this to be merged.

**Comment by malvikasharan** [comment]
@arronlacey Apology for not addressing on of your points in #104 
https://github.com/alan-turing-institute/open-community-building/pull/104#issuecomment-1196752410

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:304]: @arronlacey should we add something here to address what if a resolution can't be agreed (from #104).

Assumption is that people involved in report handling will arrive to some kind of resolution. However, we can include that if resolution is not reached, the ombudsperson will be cconsulted and any delay will be communicated with the reporter and reportee.

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:211]: @arielle-bennett thanks for agreeing to be added in the CoC. Can you please look at the CoC and the details that this role entails?

**Comment by malvikasharan** [comment]
@EKaroune and @arronlacey have agreed to included as CoC contact in this document.

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [reply CODE_OF_CONDUCT.md:214]: Approved by Arielle personally.

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [reply CODE_OF_CONDUCT.md:307]:
```suggestion
1. if a resolution is not reached, the ombudsperson will be consulted and any delay or status update for the CoC enforcement will be communicated with the reporter and reportee.
```

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:247]:
```suggestion
**15.1.1** A meeting with the reportee will be scheduled where the person handling the report will discuss five questions:
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:31]:
```suggestion
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoers. 
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:39]:
```suggestion
  Communities of Practice that welcomes and supports members of all backgrounds 
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:62]:
```suggestion
  threats or language directed against another person, discriminatory jokes and
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:63]:
```suggestion
  language, posting sexually explicit or violent material, posting (or
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:65]:
```suggestion
  (“doxing”), personal insults, especially those using racist or sexist terms,
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:66]:
```suggestion
  unwelcome sexual attention, advocating for or encouraging, any of the above
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:67]:
```suggestion
  behaviour, and repeated harassment of others. In general, if someone asks you to
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:136]:
```suggestion
5. Seeks to right the wrongs that have been done so that victims feel safe and valued, and wrongdoers feel restored to the team/community.
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:145]:
```suggestion
5. Engages in collaborative problem solving.
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:147]:
```suggestion
7. Enhances responsibility for actions and attitudes of all involved.
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:300]:
```suggestion
1. an apology to the person harmed or the team/community as a whole - this will be done via the person handling the report.
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md:301]:
```suggestion
1. an action plan for the reportee to improve their awareness and correct future behaviours.
```

File: CODE_OF_CONDUCT.md
Comment (vhellon) [inline CODE_OF_CONDUCT.md]:
```suggestion
1. reportee will have no interaction with the reporter, mediating interaction between reporter and reportee is not in scope of this Code of Conduct.
```

**Review by vhellon — COMMENTED** [review]
All looks good to me- have just corrected some typos

**Comment by RaoOfPhysics** [comment]
LGTM! 🚀

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [inline CODE_OF_CONDUCT.md:6]:
```suggestion
participants within the team as well as this online repository (and the ideas, projects, and collaborations that might emerge from it).
```
Just adding this for clari

**Comment by Arielle-Bennett** [comment]
This all looks good to me, happy to be ombudsperson for the CoC ✨

**Comment by malvikasharan** [comment]
@vhellon @aleesteele @RaoOfPhysics @Arielle-Bennett @arronlacey Thank you all for review and comments. 🎊

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:309]:
```suggestion

**16.1** After the Resolution is Agreed Upon
```

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:288]:
```suggestion
### **15.2.2** If the reportee is not a Turing Employee
```

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:271]:
```suggestion
In this case, the report will be passed to the HR and/or appropriate personnel under the Turing policies and their policy related enforcement mechanism will take precedent over this Code of Conduct enforcement.
```

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [inline CODE_OF_CONDUCT.md:29]:
```suggestion
“restorative practices” approach to handle the situation. Restorative practices have
```

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [inline CODE_OF_CONDUCT.md]:
```suggestion
repairing the harm done to people and the relationships rather than on punishing offenders or wrongdoer. This approach emerged from studies of and in reaction to incarceration and punishment within criminal justice systems, and has its roots in indigenous belief systems.
```
I suggest this addition for greater context & intellectual lineage, especially for those that might not be familiar with the approach.

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [reply CODE_OF_CONDUCT.md]: Is it 'welcome and support' since communities of practice is plural?

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [inline CODE_OF_CONDUCT.md:88]:
```suggestion
1. Accepting responsibility for harm done to people and/or communities, correcting our mistakes,
```

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [inline CODE_OF_CONDUCT.md:85]: The numbers are displayed as a series of 1's, not sure why on my end.

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [inline CODE_OF_CONDUCT.md:89]: I wonder if it would be worth adding in quotes: 
```suggestion
  and learning from the experience to avoid future harm (following the 'regenerative justice' framework)
```

**Review by aleesteele — APPROVED** [review]

**Review by eirini-zormpa — APPROVED** [review]

File: CODE_OF_CONDUCT.md
Comment (sgibson91) [reply CODE_OF_CONDUCT.md:85]: This is a markdown "hack". If you look at the rendered copy, you'll see that the numbers progress as you'd expect. This means that authors don't have to keep track of how many bullets they have, and can also help mitigate issues where instructions might say "refer to step 3", but then a new PR adds a new step before so step 3 isn't step 3 anymore, etc.

File: CODE_OF_CONDUCT.md
Comment (sgibson91) [inline CODE_OF_CONDUCT.md:141]:
```suggestion
1. Acknowledges that trust, safe spaces and relationships are central to community building.
```

?

**Review by sgibson91 — APPROVED** [review]

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [reply CODE_OF_CONDUCT.md:89]: I would avoid adding new concept unless strictly necessary. Has this term been mentioned before? If not, do we need to mention it at all?

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [reply CODE_OF_CONDUCT.md:41]: The 'commitment' section is supposed to be exhaustive. This one is not exhaustive, instead we highlight the top protected characteristics. This can be kept identical if you suggest.

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [reply CODE_OF_CONDUCT.md:85]: Thanks Sarah for addressing this!

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [reply CODE_OF_CONDUCT.md]: Slight rewording suggestion for this context:
> Although the 'restorative justice' term was in 1970 in the context of social work and justice systems, restorative practice has its roots in indigenous practices employed in cultures all over the world.

**Comment by malvikasharan** [comment]
Thanks @aleesteele @eirini-zormpa and @sgibson91 for the review.

@aleesteele and @eirini-zormpa, there are some unresolved comments. Can you please have a look when you have time next week? Especially to check if my response addresses your comments/suggestions.

File: CODE_OF_CONDUCT.md
Comment (eirini-zormpa) [reply CODE_OF_CONDUCT.md:41]: No, that's okay! I would maybe just remove "colour" from this list as I think it doesn't add anything to race and reads as perhaps older terminology? Do let me know if I'm missing something though 😊

Comment (malvikasharan) [reply CODE_OF_CONDUCT.md:41]: Now that inconsistency is definitely bothering me. Let me suggest some consistent change.

Comment (malvikasharan) [reply CODE_OF_CONDUCT.md]: I have merged "our commitment" with "diversity statement" into single clause 2: https://github.com/alan-turing-institute/open-community-building/blob/code-of-conduct/CODE_OF_CONDUCT.md#2-our-commitment-and-diversity-statement

To specifically address the inconsistency issue, I have now updated 4.2 that references clause 2.
Let me know if this actually works better.

**Review by RaoOfPhysics — APPROVED** [review]

File: CODE_OF_CONDUCT.md
Comment (eirini-zormpa) [reply CODE_OF_CONDUCT.md]: I like that!

Comment (malvikasharan) [reply CODE_OF_CONDUCT.md]: Great! Thank you Eirini.

**Comment by malvikasharan** [comment]
Hi @aleesteele can you please respond to the unresolved comments?

File: CODE_OF_CONDUCT.md
Comment (malvikasharan) [inline CODE_OF_CONDUCT.md:33]:
```suggestion
our best to right the wrong.

Although the 'restorative justice' term was coined in 1970 in the context of social work and justice systems, restorative practice has its roots in indigenous practices employed in cultures all over the world.
```

**Comment by malvikasharan** [comment]
I have verbal approval from Arron and Emma from our strategy meeting. Posting here for the record.

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [reply CODE_OF_CONDUCT.md:33]: Looks great!

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [reply CODE_OF_CONDUCT.md:94]: Thanks so much @sgibson91 for the clarification! 👍

File: CODE_OF_CONDUCT.md
Comment (aleesteele) [reply CODE_OF_CONDUCT.md:98]: I agree! Especially with the addition of the line above about regenerative justice, I don't think this is a necessary addition anymore.

**Comment by malvikasharan** [comment]
Merging this!! Thank you all! <3

**Pull request merged**
